File: front-end/lib/src/models/Personne.dart
import 'package:arosaje/src/models/Role.dart';

class Personne {
  final int id;
  final String adresse;
  final int cp;
  final String email;
  final String mdp;
  final String nom;
  final String prenom;
  final String ville;
  final Role role;

  const Personne({
    required this.id,
    required this.adresse,
    required this.cp,
    required this.email,
    required this.mdp,
    required this.nom,
    required this.prenom,
    required this.ville,
    required this.role,
  });

  static List<Personne> listFromJson(List<dynamic> parsedJson) {

    List<Personne> personne = <Personne>[];
    for (var personne in  parsedJson) {
      personne.add(Personne.fromJson(personne));
    } 
    return personne;
  }

  factory Personne.fromJson(Map<String, dynamic> json) {
    return Personne(
      id: json['id'],
      adresse: json['adresse'],
      cp: json['cp'], 
      email: json['email'],
      mdp: json['mdp'],
      nom: json['nom'],
      prenom: json['prenom'],
      ville: json['ville'],
      role: Role.fromJson(json['role']),
    );
  }
}

File: site_web_arosaje_ball/src/models/Personne.ts
import Role from "@/models/Role";

export default class Personne {
  private id: number;
  private adresse: string;
  private cp: number;
  private email: string;
  private mdp: string;
  private nom: string;
  private prenom: string;
  private ville: string;
  private role: Role;


  constructor(id: number, adresse: string, cp: number, email: string, mdp: string, nom: string, prenom: string, ville: string, role: Role) {
    this.id = id;
    this.adresse = adresse;
    this.cp = cp;
    this.email = email;
    this.mdp = mdp;
    this.nom = nom;
    this.prenom = prenom;
    this.ville = ville;
    this.role = role;
  }


  getId(): number {
    return this.id;
  }

  setId(value: number) {
    this.id = value;
  }

 getAdresse(): string {
    return this.adresse;
  }

 setAdresse(value: string) {
    this.adresse = value;
  }

  getCp(): number {
    return this.cp;
  }

  setCp(value: number) {
    this.cp = value;
  }

  getEmail(): string {
    return this.email;
  }

  setEmail(value: string) {
    this.email = value;
  }

  getMdp(): string {
    return this.mdp;
  }

  setMdp(value: string) {
    this.mdp = value;
  }

  getNom(): string {
    return this.nom;
  }

  setNom(value: string) {
    this.nom = value;
  }

  getPrenom(): string {
    return this.prenom;
  }

  setPrenom(value: string) {
    this.prenom = value;
  }

  getVille(): string {
    return this.ville;
  }

  setVille(value: string) {
    this.ville = value;
  }

  getRole(): Role {
    return this.role;
  }

  setRole(value: Role) {
    this.role = value;
  }
}
